test/testdoT.js

```diff
@@ -61,4 +61,15 @@ describe('doT', function(){
 		});
 	});
 
+	describe('whitespace stripping', function() {
+		function assert_match(actual, expected) {
+			assert(actual.match(expected), JSON.stringify(actual) + " did not match " + expected);
+		}
+		it('should replace stretches of whitespace with one whitespace', function() {
+			assert_match(doT.template('<div>    </div>')(), /^<div>\s<\/div>$/);
```

Comment:
It is an effort not to overspecify; the current code will accept any valid whitespace, as long as it is just one character. That is the functional requirement.

I could accept the exact string match as well, because this relaxation is not worth any confusion :)